//
// Copyright (C) 2024 EA group inc.
// Author: Jeff.li lijippy@163.com
// All rights reserved.
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU Affero General Public License as published
// by the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU Affero General Public License for more details.
//
// You should have received a copy of the GNU Affero General Public License
// along with this program.  If not, see <https://www.gnu.org/licenses/>.
//
//


syntax="proto2";
package halakv;

option cc_generic_services = true;

message KvRequest {
      required string key = 1;
      optional string value = 2;
};

message KvResponse {
      required int32 code = 1;
      required string message = 2;
      optional string value = 3;
};

service KvService {
      rpc set(KvRequest) returns (KvResponse);
      rpc get(KvRequest) returns (KvResponse);
      rpc remove(KvRequest) returns (KvResponse);
};
